TTHAnalysis/python/tools/higgsDiffCompTTH.py

```diff
@@ -103,10 +116,56 @@ def analyze(self, event):
         pTTrueGen       = getattr(event,'%spTTrueGen'%self.label)
         pTTrueGenPlusNu = getattr(event,'%spTTrueGenPlusNu'%self.label)
         
+        allLeps     = Collection(event,"LepGood","nLepGood")
+        nFO      = getattr(event,"nLepFO_Recl")
+        selLeps    = getattr(event,"iLepFO_Recl")
+        leps   = [allLeps[selLeps[i]] for i in xrange(nFO)]
         thejets     = [x for x in Collection(event,"JetSel_Recl","nJetSel_Recl")]
         thejetsNoB     = [j for j in thejets if j.btagDeepB<btagvetoval]
+        thejetsmore = [x for x in Collection(event,"Jet")]
+        thejetsphieta = [[j.phi, j.eta] for j in thejets]
+        thejetsmoreskimmed = [j for j in thejetsmore if [j.phi,j.eta] in thejetsphieta]
+
+        # Get right & wrong lepton index
+        rightlep = -1
+        wronglep = -1
+        if (len(QFromWFromH)==2 and len(LFromWFromH)==1):
+            ilist=[i[0] for i in sorted(enumerate(leps),key=lambda x:x[1].p4().DeltaR(LFromWFromH[0]))]
+            rightlep = ilist[0]
+            wronglep = ilist[-1] # last element in list
+
+        # Get higgs mass by combining gen quarks with right & wrong leptons
+        mHrightlep = -99
+        mHwronglep = -99
+        if len(QFromWFromH)==2 and len(LFromWFromH)==1:
+            mHrightlep = (leps[rightlep].p4()+QFromWFromH[0]+QFromWFromH[1]).M()
+            mHwronglep = (leps[wronglep].p4()+QFromWFromH[0]+QFromWFromH[1]).M()
         
         for jesLabel in self.systsJEC.values():
+            htt_PtTop = -99
```

Comment:
This block should be removed: the information is already produced when running higgsDiffRecoTTH.py: if you need additional variables please add them there.
Here you then access the previously produced diffRecoTTH friend tree.